Python/ast_opt.c

```diff
@@ -631,6 +632,63 @@ fold_compare(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
     return 1;
 }
 
+/* bpo-44501: Pack call arguments into a constant tuple if they are
+ *            all constants. The limits are important since the optimization
+ *            shows significance after 3 or more elements, and we don't want
+ *            to create giant tuples. */
+
+#define AST_MIN_CALL_FOLD 3
+#define AST_MAX_CALL_FOLD 100
+
+static int
+fold_call(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
+{
+    assert(node->kind == Call_kind);
+
+    asdl_expr_seq *args = node->v.Call.args;
+    Py_ssize_t nargs = asdl_seq_LEN(args);
+    Py_ssize_t nkeywords = asdl_seq_LEN(node->v.Call.keywords);
+
+    if (nkeywords || !nargs) {
+        return 1;
+    }
+
+    if (nargs < AST_MIN_CALL_FOLD || nargs > AST_MAX_CALL_FOLD) {
+        return 1;
+    }
+
+    for (Py_ssize_t i = 0; i < nargs; i++) {
+        expr_ty elt = asdl_seq_GET(args, i);
+        if (elt->kind != Constant_kind) {
+            return 1;
+        }
+    }
+
+    PyObject *args_tuple = make_const_tuple(args);
+    if (!args_tuple || _PyArena_AddPyObject(arena, args_tuple) < 0) {
+        Py_XDECREF(args_tuple);
+        return 0;
+    }
+
+    assert(asdl_seq_LEN(args) >= 1);
```

Comment:
It is already tested above. And the code below uses `nargs`, not `asdl_seq_LEN(args)`.

Reply:
We can safely drop this assert.